src/main/flight/interpolated_setpoint.c

```diff
@@ -29,24 +29,43 @@
 #include "flight/interpolated_setpoint.h"
 
 static float setpointDeltaImpl[XYZ_AXIS_COUNT];
-static float prevSetpointDeltaImpl[XYZ_AXIS_COUNT];
 static float setpointDelta[XYZ_AXIS_COUNT];
+static float holdCount[XYZ_AXIS_COUNT];
 
+typedef struct laggedMovingAverageCombined_u {
+     laggedMovingAverage_t filter;
+     float buf[4];
+} laggedMovingAverageCombined_t;
 
+laggedMovingAverageCombined_t  setpointDeltaAvg[XYZ_AXIS_COUNT];
+ 
 static float prevSetpointSpeed[XYZ_AXIS_COUNT];
+static float prevAcceleration[XYZ_AXIS_COUNT];
 static float prevRawSetpoint[XYZ_AXIS_COUNT];
-static float prevSetpointAcceleration[XYZ_AXIS_COUNT];
-
+static bool bigStep[XYZ_AXIS_COUNT];
 
 // Configuration
 static float ffMaxRateLimit[XYZ_AXIS_COUNT];
 static float ffMaxRate[XYZ_AXIS_COUNT];
 
 void interpolatedSpInit(const pidProfile_t *pidProfile) {
     const float ffMaxRateScale = pidProfile->ff_max_rate_limit * 0.01f;
+    int j = 1;
+    switch(pidProfile->ff_interpolate_sp){
+        case FF_INTERPOLATE_AVG2:
+            j = 2;
+            break;
+        case FF_INTERPOLATE_AVG3:
+            j = 3;
+            break;
+        case FF_INTERPOLATE_AVG4:
+            j = 4;
+            break;
+    }
```

Comment:
We can just use a `uint8_t` value with an appropriate input range (1 to 4), and avoid the switch statement.

Reply:
Thanks, much better!